specification/keyvault/data-plane/TypeSpec/main.tsp

```diff
@@ -0,0 +1,47 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "./routes.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+using Azure.Core;
+
+/**
+ * The key vault client performs cryptographic key operations and vault operations
+ * against the Key Vault service.
+ */
+@useAuth(
+  OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit,
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/authorize",
+      scopes: ["https://vault.azure.net/.default"],
+    }
+  ]>
+)
+@service({
+  title: "Key Vault",
+})
+@versioned(Versions)
+@server(
+  "{vaultBaseUrl}",
+  "The key vault client performs cryptographic key operations and vault operations against the Key Vault service.",
+  {
+    vaultBaseUrl: url,
+  }
+)
+namespace KeyVault;
+
+/**
+ * The available API versions.
+ */
+enum Versions {
+  /**
+   * The 7.6-preview.1 API version.
+   */
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  `v7.6_preview.1`: "7.6-preview.1",
```

Comment:
Note: this has already shipped, but as a comparison it's good to include. Might be worth adding "7.6-preview.2", though.